autoload/lsp/ui/vim/utils.vim

```diff
@@ -19,7 +19,7 @@ function! lsp#ui#vim#utils#locations_to_loc_list(result) abort
 
                 if has_key(l:cache, l:path)
                     let l:text = l:cache[l:path][l:index]
-                elseif l:bufnr >= 0
+                elseif l:bufnr >= 0 && !empty(getbufline(l:bufnr, 1, 2))
```

Comment:
Could you rewrite it a bit so we don't call `getbufline` twice?

Reply:
`getbufline` support use file path too, we do not need `l:bufnr` any more.